otel-extension/src/main/scala/io/scalac/mesmer/agent/Agent.scala

```diff
@@ -0,0 +1,103 @@
+package io.scalac.mesmer.agent
+
+import java.lang.instrument.Instrumentation
+
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer
+import net.bytebuddy.agent.builder.AgentBuilder
+import net.bytebuddy.description.`type`.TypeDescription
+import net.bytebuddy.dynamic.DynamicType
+import net.bytebuddy.matcher.ElementMatcher
+import org.slf4j.LoggerFactory
+
+import scala.jdk.CollectionConverters.SeqHasAsJava
+
+import io.scalac.mesmer.agent.util.i13n
+import io.scalac.mesmer.agent.util.i13n.InstrumentationDetails
+
+final case class Agent private (private[agent] val instrumentations: Set[AgentInstrumentation]) extends {
```

Comment:
Just a note for the future: as you noted in some previous reviews, it's hard to review copied code. So we should mitigate this situation in the future by not doing both moving and changing code in the same PRs. This way we see what PR exactly fixes the problem and what PRs are just pure refactoring.